### ISSUE TITLE
Change rate from schedule function

### ISSUE THREAD

**Issue by Thandriel** [cazala/synaptic#63]
It would be nice to have possibility to change rate value from inside the schedule function.
Example:
```javascript
var lastError = 0;
var trainer = new synaptic.Trainer(network);
trainer.train(trainingSet, {..., schedule: {
  every: 1,
  do: function(data){
    if(lastError - data.error > 0){
      trainer.rate *= 1.2;
    }else{
      trainer.rate *= 0.9;
    }
    lastError = data.error;
  }
}});
```
Values of data.rate seem to be read only and trainer.rate does not affect the rate used in training process (E.g. data.rate doesn't change)

**Comment by cazala** [comment]
good catch, that's because we create a private var scoped inside the .train() function called currentRate, and we use that one for the whole training (the reason of this is that the rate can be variable, like when you pass an array of floats instead of just a float to the trainer options) but it would be an easy fix to assign the value of Trainer#rate if it is a number to the currentRate after each iteration, that way you would be able to modify the rate during the training.

I'm already thinking of re-writing a big part of the trainer, specially because we have a lot of duplicated code among the `train` and `workerTrain` methods so I wanna rewrite those in order to make it more DRY, I ll take this into account once I get to start with that :)
